<launch>

  <arg name="point_cloud" />

  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

  <!-- Segment the plane -->
  <node pkg="nodelet" type="nodelet" name="plane_fitting" args="load pcl/SACSegmentation pcl_manager" output="screen">
    <remap from="~input"   to="$(arg point_cloud)" />
    <rosparam>
      model_type: 1
      distance_threshold: 0.05
      min_inliers: 10
      max_iterations: 1000
      optimize_coefficients: true
      method_type: 2
      <!--
      ("probability", double_t, 0, "The desired probability of choosing at least one sample free from outliers", 0.99, 0.5, 0.99) 
      -->
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="extract_plane_indices" args="load pcl/ExtractIndices pcl_manager" output="screen">
    <remap from="~input"   to="$(arg point_cloud)" />
    <remap from="~indices" to="/plane_fitting/inliers" />
    <param name="negative" value="false" />
  </node>
</launch>

